buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/LicenseHeadersTask.groovy

```diff
@@ -33,8 +35,27 @@ import java.nio.file.Files
  */
 public class LicenseHeadersTask extends AntTask {
 
+    @OutputFile
+    File reportFile = new File(project.buildDir, 'reports/licenseHeaders/rat.log')
+
+    private List<FileCollection> javaFiles
+
     LicenseHeadersTask() {
         description = "Checks sources for missing, incorrect, or unacceptable license headers"
+        // Delay resolving the dependencies until after evaluation so we pick up generated sources
+        project.afterEvaluate {
+            List<FileCollection> javaFiles = project.sourceSets.collect({it.allJava})
+            setJavaFiles(javaFiles)
```

Comment:
You could also create a local before the afterEvaluate so it gets captured?
```
LicenseHeadersTask task = this
```

Also note that this member var should be pkg private or protected (not sure which), since the closure is really an anonymous class.